src/graphql/root.rs

```diff
@@ -395,7 +395,14 @@ graphql_object!(Query: Context |&self| {
 
     /// Find all the message types we know about
     field messageTypes(&executor) -> FieldResult<Vec<String>>{
-        Err("Not implemented")?
+        let connection = executor.context().connection.lock()?;
+        let query = messages_table
+            .select(messages_content_type)
+            .filter(messages_content_type.is_not_null())
+            .distinct()
+            .load::<Option<String>>(&(*connection))?;
```

Comment:
I'd like to avoid the `Option<>` part here, since I already filtered for `is_not_null()` above. Not sure if diesel can do this though.

Reply:
> Not sure if diesel can do this though.

Yeah, I spent some time looking. There doesn't seem to be a nice way to do it. I think it's ok though. I can see why they might not want to expose it in the api, it might be a bit of a foot-gun.